library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt

```diff
@@ -137,32 +134,34 @@ class ChuckerInterceptor @JvmOverloads constructor(
     /**
      * Processes a [ResponseBody] and populates corresponding fields of a [HttpTransaction].
      */
-    private fun processResponseBody(response: Response, responseBody: ResponseBody?, transaction: HttpTransaction) {
-        getNativeSource(response)?.use { source ->
-            source.request(java.lang.Long.MAX_VALUE)
-            val buffer = source.buffer()
-            var charset: Charset = UTF8
-            val contentType = responseBody?.contentType()
-            if (contentType != null) {
-                try {
-                    charset = contentType.charset(UTF8) ?: UTF8
-                } catch (e: UnsupportedCharsetException) {
-                    return
-                }
+    private fun processResponseBody(response: Response, responseBody: ResponseBody, transaction: HttpTransaction) {
+        val contentType = responseBody.contentType()
+        val charset: Charset = contentType?.charset(UTF8) ?: UTF8
+        val contentLength = responseBody.contentLength()
+
+        val source = responseBody.source()
+        source.request(Long.MAX_VALUE) // Buffer the entire body.
+        var buffer = source.buffer()
+
+        if (CONTENT_ENCODING_GZIP.equals(response.headers()[CONTENT_ENCODING], ignoreCase = true)) {
+            GzipSource(buffer.clone()).use { gzippedResponseBody ->
+                buffer = Buffer()
+                buffer.writeAll(gzippedResponseBody)
             }
-            if (io.isPlaintext(buffer)) {
-                val content = io.readFromBuffer(buffer, charset, maxContentLength)
-                transaction.responseBody = content
-            } else {
-                transaction.isResponseBodyPlainText = false
+        }
 
-                val isImageContentType = (transaction.responseContentType?.contains(CONTENT_TYPE_IMAGE) == true)
+        if (io.isPlaintext(buffer)) {
+            transaction.isResponseBodyPlainText = true
+            if (contentLength != 0L) {
+                transaction.responseBody = buffer.clone().readString(charset)
+            }
+        } else {
+            transaction.isResponseBodyPlainText = false
 
-                if (isImageContentType && buffer.size() < MAX_BLOB_SIZE) {
-                    transaction.responseImageData = buffer.readByteArray()
-                }
+            val isImageContentType = (transaction.responseContentType?.contains(CONTENT_TYPE_IMAGE) == true)
```

Comment:
Headers are case-insensitive. I know that it is hard to find an API, which sends `Image` or somehow else, except `image`, but let's follow the standard.
https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2

Reply:
Yeah, but it is just one word to add here.
```suggestion
            val isImageContentType = (transaction.responseContentType?.contains(CONTENT_TYPE_IMAGE, ignoreCase = true) == true)
```